library/old_protocol/serialize.cpp

```diff
@@ -84,21 +84,53 @@ int serialize_lookup_request(dnet_net_state *st, std::unique_ptr<n2_request> msg
 
 int serialize_lookup_response(dnet_net_state *st, std::unique_ptr<n2_message> msg_in,
                               std::unique_ptr<n2_serialized> &out_serialized) {
-	auto impl = [&] {
-		auto &msg = static_cast<lookup_response &>(*msg_in);
+	auto &msg = static_cast<lookup_response &>(*msg_in);
 
-		msgpack::sbuffer msgpack_buffer;
-		msgpack::pack(msgpack_buffer, msg);
+	auto impl = [&] {
+		data_pointer data;
+		size_t size = 0;
+
+		if (msg.cmd.cmd == DNET_CMD_LOOKUP_NEW) {
```

Comment:
We should discuss how must the serializer get info about request context. Current behavior seems ok as a temporary solution. But after we finalize refactoring, `lookup_response` won't have info about `cmd.cmd`. All the context must be embedded into `repliers`. So, better behavior (not temporary in my opinion) = to have personal `serializer`/`deserializer`/`lookup_*_replier`/`old_protocol::translate_lookup_*_request` for each (old/new) command. Execution must converge at protocol_interface::on_request. I can describe details by voice.

Reply:
The main idea: lookup/lookup_new serialization is old_protocol private deal. When we try to manage protocol from handlers (e.g. specifying `cmd.cmd` explicitly), we break abstration, because we look back on concrete protocol implementation.

Reply:
As I know, we have plans to get rid of commands without `_NEW` suffix.

Reply:
bdeefb0

Reply:
bdeefb0